core/src/components/tab-button/tab-button.ionic.scss

```
@@ -86,3 +88,52 @@
 :host(.tab-button-shape-rectangular) {
   @include globals.border-radius(globals.$ion-border-radius-0);
 }
+
+// Tab Button: Badge
+// --------------------------------------------------
+
+:host ::slotted(ion-badge) {
+  @include globals.position(null, null, null, calc(50% + globals.$ion-scale-300));
+}
+
+:host ::slotted(ion-badge.badge-vertical-top) {
+  @include globals.position(-2px, null, null, null);
```

Since we don't change any of the other value, we can depend on the defaults. I recommend reviewing how `position` works.
```suggestion
  @include globals.position(-2px);
```

---

That's why I think my first approach is the best approach... Here we should remove default values, but [here](https://github.com/ionic-team/ionic-framework/blob/1552104c15253e887976e05f1758b83c5e2a2487/core/src/components/tab-button/tab-button.ionic.scss#L96) we must keep them...

---

This is just how the Framework has been doing it. You are welcome to open a discussion with the team about changing the structure.